gateway-js/src/loadSupergraphSdlFromStorage.ts

```diff
@@ -43,12 +43,20 @@ export async function loadSupergraphSdlFromStorage({
   apiKey,
   endpoint,
   fetcher,
+  ifAfterId,
+  currentSupergraphSdl,
 }: {
   graphRef: string;
   apiKey: string;
   endpoint: string;
   fetcher: typeof fetch;
+  ifAfterId: string | null;
+  currentSupergraphSdl: string | null;
```

Comment:
Just proposing an alternative - it may not be any better or worse than what you've already got!

I'm not sure this function needs to have awareness of the current sdl. We could instead do the following:
* only pass in an `inAfterId` if we have both `this.compositionId` and `this.supergraphSdl` (though I can't imagine a case we'd have one without the other [ref](https://github.com/apollographql/federation/blob/version-0.x/gateway-js/src/index.ts), so maybe just the `compositionId` is sufficient)
* in the `routerConfig.__typename === 'Unchanged'` case, return `null` instead of the previous values, and handle the `null` case at the call site

And while `ifAfterId` does make sense as a query arg, the naming is a bit more of an implementation detail of our API. I think it makes sense to name the arg coming into this function either `compositionId` or `currentCompositionId` to be more inline with the naming of things at the call site.

Reply:
I don't really like my solution to this problem of not overwriting the existing schema, but I don't see how to do what you suggest without changing `SupergraphSdlUpdate` to look like this:

```
export interface SupergraphSdlUpdate {
  id: string | null;
  supergraphSdl: string | null;
}
```

Which leads to lots of other changes to make TypeScript happy with types.

I tried dealing with the null results in `isSupergraphSdlUpdate()`.

Reply:
Resolved by c1ecd02